libmei/tools/parseschema2.py

```diff
@@ -0,0 +1,328 @@
+# -- coding: utf-8 --
+
+# Copyright (c) 2011-2022 Andrew Hankinson, Alastair Porter, and Others
+
+# Permission is hereby granted, free of charge, to any person obtaining
+# a copy of this software and associated documentation files (the
+# "Software"), to deal in the Software without restriction, including
+# without limitation the rights to use, copy, modify, merge, publish,
+# distribute, sublicense, and/or sell copies of the Software, and to
+# permit persons to whom the Software is furnished to do so, subject to
+# the following conditions:
+
+# The above copyright notice and this permission notice shall be
+# included in all copies or substantial portions of the Software.
+
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+# NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+# LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+# OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+# WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+import logging
+import re
+import shutil
+from argparse import ArgumentParser
+from pathlib import Path
+
+from lxml import etree
+
+import cpp
+from utilities import elapsedtime
+
+lg = logging.getLogger("schemaparser")
+f = logging.Formatter("[%(asctime)s] [%(levelname)8s] %(message)s (%(filename)s:%(lineno)s)")
+h = logging.StreamHandler()
+h.setFormatter(f)
+
+lg.setLevel(logging.DEBUG)
+lg.addHandler(h)
+
+# globals
+TEI_NS = {"tei": "http://www.tei-c.org/ns/1.0"}
+TEI_RNG_NS = {"tei": "http://www.tei-c.org/ns/1.0",
+              "rng": "http://relaxng.org/ns/structure/1.0"}
+NAMESPACES = {'xml': 'http://www.w3.org/XML/1998/namespace',
+              'xlink': 'http://www.w3.org/1999/xlink'}
+
+WHITESPACE_REGEX: Pattern = re.compile(r"[\s\t]+")
```

Comment:
now typing.Pattern is missing

Reply:
I did that on purpose, just to see if you would notice.

Reply:
I think it's fair to say that this is still WIP (hence Draft) so not quite ready for review.